Tests/Unit/LoRaTools/MacCommandTests.cs

```diff
@@ -0,0 +1,293 @@
+            public override Cid Cid => Cid.RXParamCmd;
+            public override RXParamSetupAnswer Subject => new RXParamSetupAnswer(true, false, true);
+            public override IReadOnlyList<byte> DataBytes => new byte[] { 0b101 };
+            public override int Length => 2;
+
+            [Fact]
+            public void FromBytes_Success() => FromBytesTest(actual => actual.Cid == Cid
+                                                                    && actual.Rx1DROffsetAck
+                                                                    && !actual.Rx2DROffsetAck
+                                                                    && actual.ChannelAck,
+                                                             bytes => new RXParamSetupAnswer(new ReadOnlySpan<byte>(bytes)),
+                                                             DataBytes);
+        }
+
+        public sealed class RxParamSetupRequestTests : MacCommandTests
+        {
+            public override Cid Cid => Cid.RXParamCmd;
+            public override RXParamSetupRequest Subject => new RXParamSetupRequest(1, 2, 3);
+            public override IReadOnlyList<byte> DataBytes => new byte[] { 0b10010, 3, 0, 0 };
+            public override int Length => 5;
+
+            [Fact]
+            public void Deserializes_Correctly() =>
+                DeserializationTest<RXParamSetupRequest>(actual => actual.Cid == Cid
+                                                                && actual.Frequency == Subject.Frequency
+                                                                && actual.RX1DROffset == Subject.RX1DROffset
+                                                                && actual.RX2DataRate == Subject.RX2DataRate,
+                                                         @"{""cid"":5,""frequency"":3,""dlSettings"":18}");
+        }
+
+        public sealed class RxTimingSetupAnswerTests : MacCommandTests
+        {
+            public override Cid Cid => Cid.RXTimingCmd;
+            public override MacCommand Subject => new RXTimingSetupAnswer();
+            public override IReadOnlyList<byte> DataBytes => Array.Empty<byte>();
+            public override int Length => 1;
+        }
+
+        public sealed class RxTimingSetupRequestTests : MacCommandTests
+        {
+            public override Cid Cid => Cid.RXTimingCmd;
+            public override RXTimingSetupRequest Subject => new RXTimingSetupRequest(1);
+            public override IReadOnlyList<byte> DataBytes => new byte[] { 1 };
+            public override int Length => 2;
+
+            [Fact]
+            public void Deserializes_Correctly() =>
+                DeserializationTest<RXTimingSetupRequest>(actual => actual.Cid == Cid && actual.Settings == Subject.Settings,
+                                                          @"{""cid"":8,""settings"":1}");
+        }
+
+        [Fact]
+        public void ToBytes_Success() => ToBytes_Internal(Subject, DataBytes);
```

Comment:
nit: This and remaining methods are part of the base class `MacCommandTests` and so should be moved before the subclasses. It's somewhat odd to have subclasses sandwiched in the middle of the base class definition.

Reply:
Is it possible to avoid the base class having tests? I believe this throws off some test runners and produces warnings about duplicate tests.

Reply:
Addressed this with 973733e27f11acbf4ac0e2e12ed5326e4f435f9c. As discussed, we are keeping the tests inheriting from a base class for now.